### DISCUSSION TITLE
How do I enable new pane/tab with CWD while using mosh?

### DISCUSSION THREAD

**Discussion by abhijit-hota** [kovidgoyal/kitty#6529] [Q&A]
I've been using [Kitty's SSH features](https://sw.kovidgoyal.net/kitty/kittens/ssh/) for as long as I can remember but I recently setup [Mosh](https://mosh.org/) and I really like how it doesn't drop connections and supports roaming. 

The thing that I miss the most is creating a new pane/tab/window with the same directory (and SSHed host).

Can I make just that part work somehow?

**Comment by kovidgoyal** [comment]
No idea, I dont use mosh. You would basically need to write the
equivalent of the ssh kitten for mosh. Or alternatively, manually setup
shell integration on your remote machine as described in
https://sw.kovidgoyal.net/kitty/shell-integration

**Reply by abhijit-hota** [reply]
I see.

> alternatively, manually setup shell integration on your remote machine

If it's relevant and make things easier, I'm using Mosh to access a machine that already has `kitty` installed and set up.

**Reply by kovidgoyal** [reply]
assuming you have up-to-date kitty on the remote machine,

mosh kitten run-shell

will probably give you a shell with working shell integration
